src/cpp/session/modules/clang/RCompilationDatabase.cpp

```diff
@@ -920,12 +958,10 @@ RCompilationDatabase::CompilationConfig
    }
 
    // parse the compilation results
-   std::vector<std::string> compileArgs = parseCompilationResults(
-                                                           result.stdOut);
+   std::vector<std::string> compileArgs = parseCompilationResults(result.stdOut);
 
-   std::copy(compileArgs.begin(),
-             compileArgs.end(),
-             std::back_inserter(args));
+   // add them to the compile args
+   args.insert(args.begin(), compileArgs.begin(), compileArgs.end());
```

Comment:
Note: this change is necessary as R's headers need to come _before_ the system headers.